source/Engine.cpp

```diff
@@ -1106,6 +1111,31 @@ void Engine::Click(const Point &from, const Point &to, bool hasShift)
 	this->hasShift = hasShift;
 	isRightClick = false;
 
+	// See if this click was in the ammoBox, then see if it clicked one of the icons.
+	if(ammoBox.Contains(from))
+	{
```

Comment:
Based on the rest of this method and those related to it, this implemention should not be performed here, but rather in the method where we handle `doClickNextStep`.